sql/partitioning/partition_base.cc

```diff
@@ -207,10 +207,28 @@ bool get_part_str_for_table(const char *name, std::string &result)
                                         table_name, sizeof(table_name));
 
   // Prepare the path to the .FRM file and open the file
-  char    path[FN_REFLEN + 1];  //< Path to .FRM file
-  bool temp_table= (bool)is_prefix(table_name, tmp_file_prefix);
-  build_table_filename(path, sizeof(path) - 1, db_name, table_name, reg_ext,
-                       temp_table ? FN_IS_TMP : 0);
+  //
+  // This formerly used sql/sql_table.cc build_table_filename
+  // The problem is that the 'name' is coming into this function already encoded
+  // in fscs, but build_table_filename is expecting the components to come in
+  // encoded in the system_charset.  This results in double encoding which then
+  // causes the final path to be nonsense that points to a file that can never
+  // exist.  Since there seems to be no other obvious helper function to take
+  // this 'name' and turn it into a proper fully qualified path to the .frm file
+  // we are doing a little copy/cut-n-paste from build_table_filename that
+  // composes the full path but without the additional encoding.
+  char  path[FN_REFLEN + 1];  //< Path to .FRM file
```

Comment:
Looking at build_table_filename, it looks like it should be reasonable to extract this to a separate function to be called both here and from build_table_filename. What do you think?

Reply:
Actually @laurynas-biveinis, my first idea to fix this was to add another flag to build_table_filename (maybe something like FN_NO_ENCODE) that would force it to use my_stpnmov instead of tablename_to_filename for both the database and table name, but, I did not want to go hacking about the server code without consulting you.  Now it seems that we are both thinking along the same line.